### PR TITLE
Fix broken max length in "SNN Last 4 digits" input 

### PR DESCRIPTION
**Pull request by aldo-expensify**
cc @luacmartins: this was a change related to the new form handling, can you check that I'm not removing/breaking something that was necessary

Another property 

### Details
Explanation here: https://github.com/Expensify/App/issues/7548#issuecomment-1029164951

Remove maxLength prop from baseTextInputPropTypes
The prop is not directly used by this component

### Fixed Issues

$ https://github.com/Expensify/App/issues/7548

### Tests / QA

SO on VBA info: https://stackoverflow.com/c/expensify/questions/342/525#525

1. Launch the app
2. Log in with expensiafail account
3. Go to any Workspace - Connect Bank Account
4. Tap Connect online with Plaid
5. Select Wells Fargo Bank Account
6. Put username: user_good, password: pass_good
7. Tap Submit
8. Tap on Plaid Saving - Save and continue
9. Put all necessary information in Compony Information Page
10. Tap Save and Continue
11. In SSN box start typing as many number you are want

- [ ] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

**Comment by cead22** [comment]
Should this be tested on all platforms?

**Comment by aldo-expensify** [comment]
The prop `hint` also got added there in the same [PR](https://github.com/Expensify/App/pull/6412/files#diff-2b104b233eb98f5f85f4dd4160ecd7cc1d35a8c101006ad1269354d3a99af6b4R75):

https://github.com/Expensify/App/blob/47ec8f4ae0598e55c4d75b275ef4feeec8f14d6e/src/components/TextInput/baseTextInputPropTypes.js#L52-L53

But I don't think it is being used anywhere. Should we remove it?

**Comment by aldo-expensify** [comment]
> Should this be tested on all platforms?

I guess it should

**Comment by cead22** [comment]
I'm going off of the [slack](https://expensify.slack.com/archives/C01SKUP7QR0/p1643823942719599?thread_ts=1643767992.306689&cid=C01SKUP7QR0) [convos](https://expensify.slack.com/archives/C01SKUP7QR0/p1643767992306689) pushing for tests on all platforms on all PRs.

Also, it looks like baseTextInputPropTypes is used by TextInput, BaseTextInpupt, and DatePicker (indirectly). Can you make sure to test that none of those inputs are breaking?

**Comment by cead22** [comment]
> But I don't think it is being used anywhere. Should we remove it?

I think so

**Comment by aldo-expensify** [comment]
> Also, it looks like baseTextInputPropTypes is used by TextInput, BaseTextInpupt, and DatePicker (indirectly). Can you make sure to test that none of those inputs are breaking in your tests?

A note on this, the `maxLength` prop in the is used by the react-native component `TextInput` (same name as ours). I understand that our approach is not to put props in the definitions if they are just drilled down.

**Comment by aldo-expensify** [comment]
I'll test on other platforms now

File: src/components/TextInput/baseTextInputPropTypes.js [hunk 1]
```diff
@@ -46,9 +46,6 @@ const propTypes = {
     /** Saves a draft of the input value when used in a form */
     shouldSaveDraft: PropTypes.bool,
 
-    /** Character limit for the input */
-    maxLength: PropTypes.number,
```
Comment (luacmartins) [inline src/components/TextInput/baseTextInputPropTypes.js:50]: `maxLength` and `hint` will be used once [TextInput](https://github.com/Expensify/App/issues/7522) is refactored. We should probably keep it here and find another solution to this issue.

Comment (aldo-expensify) [reply src/components/TextInput/baseTextInputPropTypes.js:50]: Can you add it once you refactor it?

Comment (luacmartins) [reply src/components/TextInput/baseTextInputPropTypes.js:50]: Sure, but wouldn't we run into this same issue once a contributor refactors the component?

Comment (luacmartins) [reply src/components/TextInput/baseTextInputPropTypes.js:50]: Discussed 1:1 and we will remove it for now and reintroduce these props on the actual implementation.

**Comment by aldo-expensify** [comment]
Update: tested in all platforms, seems to be working fine.

**Review by luacmartins — APPROVED** [review]

**Comment by luacmartins** [comment]
Merging since it's a revert of my changes.

**Pull request merged**